models/topics.models.js

```diff
@@ -6,35 +6,41 @@ exports.selectTopics = () => {
   });
 };
 
-exports.selectArticles = () => {
-  return db
-    .query(
-      `SELECT 
-      articles.article_id, 
-      title,
-      topic,
-      articles.author,
-      articles.created_at,
-      articles.votes,
-      article_img_url, 
-      
-      COUNT(comments.comment_id) AS comment_count
-      FROM articles
-      JOIN comments ON articles.article_id = comments.article_id
-      GROUP BY articles.article_id
-      ORDER BY created_at DESC;`
-    )
-    .then((articles) => {
+exports.selectArticles = (topic) => {
+  let queryString = `SELECT 
+  articles.article_id, 
+  title,
+  topic,
+  articles.author,
+  articles.created_at,
+  articles.votes,
+  article_img_url, 
+  
+  COUNT(comments.comment_id) AS comment_count
+  FROM articles
+  JOIN comments ON articles.article_id = comments.article_id`;
+
+  if (topic) {
+    queryString += ` WHERE articles.topic = $1 GROUP BY articles.article_id ORDER BY created_at DESC;`;
```

Comment:
Some of this query string (the GROUP BY clause) should be concatenated on whether there is a topic or not so could always be added at the end to keep things a bit more DRY.